tools/pip-compile-wrapper.py

```diff
@@ -0,0 +1,300 @@
+#!/usr/bin/env python
+"""
+Helper utility to compile / upgrade requirements files from templates.
+
+This only manages dependencies between requirements sources.
+The actual compiling is delegated to ``pip-compile`` from the ``pip-tools` package.
+
+NOTE: This utility must only use stdlib imports in order to be runnable even
+      before the dev requirements are installed.
+"""
+
+import os
+import re
+import shlex
+import subprocess
+import sys
+from collections import defaultdict
+from enum import Enum
+from itertools import chain, groupby, repeat
+from operator import itemgetter
+from optparse import OptionParser
+from pathlib import Path
+from shutil import which
+from typing import Dict, Iterable, Iterator, List, Optional, Set, Tuple
+
+REQUIREMENTS_SOURCE_DEV = "requirements-dev.in"
+REQUIREMENT_RE = re.compile(r"^([^=!~<>\[ ]+)")
```

Comment:
What about using the regex from the pep (and adding the link to the code)?

https://www.python.org/dev/peps/pep-0508/#names

Reply:
Good idea.